doc/glossary/index.rst

```diff
@@ -31,3 +31,374 @@ Glossary
+        DB2 or other database.
+
+        PrivacyIDEA supports a great variety of different token types.
+        They each have different requirements concerning configuration and how
+        the authentication works. This chapter explains the authentication modes, lists the
+        supported hardware and software tokens and explains how the token types can be used
+        with privacyIDEA.
+        Tools which facilitate and automate token enrollment are found in :ref:`enrollment_tools`.
+
+        * :ref:`authentication_modes`
+        * :ref:`supported_tokens`
+        * :ref:`tokentypes`
+
+   Audit
+        The systems provides a sophisticated audit log, that can be viewed in the WebUI.
+
+        The Audit log lists all events the server registers.
+
+        .. figure:: /audit/auditlog.png
+           :width: 500
+
+           *Audit Log*
+
+        privacyIDEA comes with a default SQL audit module (see :ref:`code_audit`).
+
+        Starting with version 3.2 privacyIDEA also provides a :ref:`logger_audit` and
+        a :ref:`container_audit` which can be used to send privacyIDEA audit log messages
+        to services like splunk or logstash.
+
+   Tokenowner
+        The owner of a token is the user for whom the token was rolled out.
+
+   FailCount
+   MaxFail
+        The FailCount count the number of failed login attempts.
+
+        If the login fail counter reaches the ``MaxFail`` the user can not login
+        with this token anymore.
+
+        The administrator or help desk user can select those tokens and
+        click the button ``reset failcounter`` to reset the fail counter to zero.
+        The tokens can be used for authentication again.
+
+   SplitAtSign
+        ``splitAtSign`` defines if the username like *user@company*
+        given during authentication should
+        be split into the loginname *user* and the realm name *company*.
+        In most cases this is the wanted behaviour so this is enabled by default.
+
+        But given your users log in with email addresses like *user@gmail.com* and
+        *otheruser@outlook.com* you probably do not want to split.
+
+        How a user is related to a realm is described here: :ref:`relate_realm`
+
+        This option also affects the login via the :ref:`rest_auth`
+
+   Rollout State
+        A token can be rolled out in several steps like the 2step HOTP/TOTP token.
+        In this case the attribute ``“rollout_state”`` of the token contains certain values
+        like ``‘clientwait’ or ‘enrolled’``. This way actions can be triggered, depending
+        on the step during an enrollment process.
+
+   Custom User Attributes
+        The table ``“customuserattribute”`` is used to store additional, custom attributes for users.
+
+        A user is identified by the user_id, the resolver_id and the realm_id.
+        The additional attributes are stored in Key and Value.
+        The Type can hold extra information like e.g. an encrypted value / password.
```

Comment:
The important part is, that the privacyIDEA administrator, who is often not allowed to manage active directory, can **add** attributes to the users. 
Normally user attributes are only read from the user store.

https://www.youtube.com/watch?v=QU30r99oE3o

Reply:
done in 4c9671867f1721aeb42e3525f74855c4335a9440.